openmc/data/endf.c

```diff
@@ -0,0 +1,39 @@
+#include <stdlib.h>
+
+double cfloat_endf(const char* buffer, int n)
```

Comment:
A few suggestions here. Given this is only called from Cython I realize that the inputs are constrained, so please disregard if you think that is already sufficient, but I'm also looking out for if this were potentially reused in some other context.
* For `buffer`, check that it isn't NULL.
* For `n`, either change type to `size_t` or `unsigned int`, or check that `n >= 0`.